### In-class Activity 9 (July 30, 2026)

In [1]:
library(tidyverse)
library(moderndive)
library(car)

── Attaching core tidyverse packages ─────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ───────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Loading required package: carData


Attaching package: ‘car’


The following object is masked from ‘package:dplyr’:

    recode


The following object is masked from ‘package:purrr’:

    some




In [2]:
hiv_data <- read.table("data/hiv_data.txt", head=TRUE)
head(hiv_data)

,trarm,patid,age,weight,CD4,logRNA
,<int>,<int>,<int>,<dbl>,<dbl>,<dbl>
1,1,21360,40,130.5,350.0,2.300
2,1,82892,39,134.0,285.0,3.610
3,1,150742,34,87.7,116.5,4.790
4,1,212952,37,168.0,253.0,3.370
5,1,213004,47,230.0,197.5,4.395
6,1,213038,39,187.0,307.5,4.245


### Part A

In [3]:
poisson_model <- glm(CD4 ~ trarm + age + weight + logRNA, family=poisson, data=hiv_data)
summary(poisson_model)

Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 116.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 197.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 307.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 357.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 264.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 157.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 256.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 428.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 154.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 189.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 222.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 535.500000”
Warning message in dpois(y, mu, log = TRUE):
“non-integer x = 216.500000”
Warning message in dpois(y, mu, log = 


Call:
glm(formula = CD4 ~ trarm + age + weight + logRNA, family = poisson, 
    data = hiv_data)

Coefficients:
              Estimate Std. Error z value Pr(>|z|)    
(Intercept)  7.1383722  0.0814654  87.625  < 2e-16 ***
trarm        0.1543540  0.0181729   8.494  < 2e-16 ***
age         -0.0035945  0.0011551  -3.112  0.00186 ** 
weight      -0.0008409  0.0002070  -4.063 4.84e-05 ***
logRNA      -0.3825030  0.0138428 -27.632  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 3449.0  on 41  degrees of freedom
Residual deviance: 2534.5  on 37  degrees of freedom
AIC: Inf

Number of Fisher Scoring iterations: 4


The p-value for trarm is quite loww (`2e-16`) and is therefore below the 0.05 significance level. Therefore at a 0.05 significance level, we reject the null hypothesis that B=0. Therefore, the treatment arm is statistically significant for determining CD4 levels.

### Part B

In [4]:
poisson_model <- glm(CD4 ~ trarm + age + weight + logRNA, family = quasipoisson, data=hiv_data)
summary(poisson_model)


Call:
glm(formula = CD4 ~ trarm + age + weight + logRNA, family = quasipoisson, 
    data = hiv_data)

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  7.1383722  0.6820490  10.466 1.31e-12 ***
trarm        0.1543540  0.1521482   1.014  0.31693    
age         -0.0035945  0.0096706  -0.372  0.71224    
weight      -0.0008409  0.0017328  -0.485  0.63033    
logRNA      -0.3825030  0.1158957  -3.300  0.00214 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for quasipoisson family taken to be 70.09466)

    Null deviance: 3449.0  on 41  degrees of freedom
Residual deviance: 2534.5  on 37  degrees of freedom
AIC: NA

Number of Fisher Scoring iterations: 4


The disersion parameter for the poisson family is `70.09` when fitting the model using the quasilikelihood method. Therefore, overdispersion is a problem for the model above since it's quite far from 1.